### PR TITLE
Feature/api

### PR DESCRIPTION
**Pull request by BartoszWojdalowicz**

### CODE REVIEW

File: src/Resources/config/services/commandHandler/command_handler.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<container
+        xmlns="http://symfony.com/schema/dic/services"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://symfony.com/schema/dic/services
+        http://symfony.com/schema/dic/services/services-1.0.xsd"
+>
+    <services>
+        <defaults public="true"/>
```
Comment (christopherhero) [inline src/Resources/config/services/commandHandler/command_handler.xml:10]:
```
<defaults public="true"/>


<service ...
```

File: src/Resources/config/services/controller/controller.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            <argument id="bitbag_sylius_banner_plugin.repository.ad"
+                      type="service"
+            />
+            <argument id="bitbag.sylius_banner_plugin.operator.banners_operator"
+                      type="service"
+            />
+        </service>
+
+        <service id="bitbag.sylius_banner_plugin.controller.get_ad_banners"
+                 class="BitBag\SyliusBannerPlugin\Controller\GetAdBanners" >
```
Comment (christopherhero) [inline src/Resources/config/services/controller/controller.xml:24]:
```
    <service id="bitbag.sylius_banner_plugin.controller.get_ad_banners"
                    class="BitBag\SyliusBannerPlugin\Controller\GetAdBanners"
    >
    

```

File: src/Resources/config/services/controller/controller.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services
+           http://symfony.com/schema/dic/services/services-1.0.xsd"
+>
+    <services>
+        <defaults public="true"/>
+
+        <service class="BitBag\SyliusBannerPlugin\Controller\GetAdsBanners"
+                 id="bitbag.sylius_banner_plugin.controller.get_ads_banners"
+                 public="true"
```
Comment (christopherhero) [inline src/Resources/config/services/controller/controller.xml:13]: `public="true"` not needed

File: src/Resources/config/services/serializer/serializer.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services
+           http://symfony.com/schema/dic/services/services-1.0.xsd"
+>
+    <services>
+        <service id="bitbag.sylius_banner_plugin.serializer.banner_serializer"
+                 class="BitBag\SyliusBannerPlugin\Serializer\BannerSerializer"
+                 >
```
Comment (christopherhero) [inline src/Resources/config/services/serializer/serializer.xml:11]:
```
    <service id="bitbag.sylius_banner_plugin.serializer.banner_serializer"
             class="BitBag\SyliusBannerPlugin\Serializer\BannerSerializer"
    >
```

**Review by christopherhero — CHANGES_REQUESTED** [review]

File: src/Resources/config/api_resources/GetAdBanners.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="https://api-platform.com/schema/metadata
+           https://api-platform.com/schema/metadata/metadata-2.0.xsd"
+>
+    <resource class="BitBag\SyliusBannerPlugin\Command\GetAdBanners" shortName="AdBanners">
+        <attribute name="validation_groups">sylius</attribute>
+
+        <itemOperations>
+            <itemOperation name="get">
+                <attribute name="controller">ApiPlatform\Core\Action\NotFoundAction</attribute>
```
[the remark this answers is not in the document]
Comment (BartoszWojdalowicz) [reply src/Resources/config/api_resources/GetAdBanners.xml:13]: There was a problem when I returned a collection of banners but the `get banner` endpoint didn't exist. So I found solution on the internet with the disabled get endpoint.

**Pull request merged**